_posts/2015-08-06-renaming-tables-with-zero-downtime.md

```diff
@@ -0,0 +1,80 @@
+---
+layout: with-comments
+title: "Renaming MySQL Tables With Zero Downtime"
+author: Marc Zych
+author_url: https://github.com/marczych
+summary: We found a way to rename MySQL tables with zero downtime by using
+         updatable views and atomic rename table commands.
+---
+
+
+Sometimes you end up with tables in your database schema that just aren't named what you want anymore.
+At first glance, renaming them without any downtime isn't feasible because updating the code that uses the table at the same instant that the table is renamed isn't easily accomplished.
+
+We recently came across a few database tables in our schema that were begging to be renamed.
+Bringing our site down for several minutes to run the migrations and deploy the updated code was unacceptable so I started investigating to come up with a solution.
+
+In this example, we will be renaming `people` to `users`.
+
+# Using a View
+
+The first thing I came up with was using [MySQL Views] to alias the table so it would be accessible by both names (`people` and `users`).
+I came to find out that [many MySQL Views are updatable and insertable][updatable views]:
+
+> Some views are updatable and references to them can be used to specify tables to be updated in data change statements. That is, you can use them in statements such as UPDATE, DELETE, or INSERT to update the contents of the underlying table.
+
+This is super cool because it means that _all_ of the queries, reads and writes, can be done against the new name.
```

Comment:
Em dashes instead of commas, perhaps? And maybe move the italics around? "it means that all of the queries―reads _and_ writes―can be done against the new name."